lib/ts/recipe/session/recipe.ts

```diff
@@ -28,18 +28,39 @@ export default class Session extends RecipeModule<unknown, unknown, unknown, any
     static instance?: Session;
     static RECIPE_ID = "session";
 
-    private eventListeners = new Set<(ctx: RecipeEvent) => void>();
+    private eventListeners = new Set<(ctx: RecipeEventWithSessionContext) => void>();
 
     constructor(config: ConfigType) {
         super(config);
 
-        if (config.onHandleEvent !== undefined) {
-            this.addEventListener(config.onHandleEvent);
-        }
-
         sessionSdk.init({
             ...config,
-            onHandleEvent: this.onHandleEvent,
+            onHandleEvent: async (event) => {
+                if (config.onHandleEvent !== undefined) {
+                    config.onHandleEvent(event);
+                }
+                if (event.action === "SESSION_CREATED" || event.action === "REFRESH_SESSION") {
```

Comment:
I'm moving this to a separate method to maintain one level of abstraction in this function.

Reply:
You really don't need to.. this refactoring serves no real purpose.. please don't over-optimize on internal code refactoring. It may introduce bugs, and it wastes time.